### PR TITLE
nextgen-maxrate [not ready]

### PR DESCRIPTION
**Pull request by tcql**
*Moved from #56 cc @morganherlocker*

- add 404 handling for missing tiles (reduce is called but with empty tile data to simulate an empty vector tile). The current behavior is to return an error on 404.
- I started refactoring how tiles are passed to workers to allow for rate limiting, but I'm still figuring out some scoping issues (tileStream yanks the scope from TileHandler, which obscures the state).

cc @mourner @tcql

### CODE REVIEW

**Comment by mourner** [comment]
Why create a separate TileHandler class? It looks like unnecessary complexity at this point.
Also, is there a reason to implement rate limiting for anything except remote sources?

File: src/remote.js [hunk 1]
```diff
@@ -23,7 +23,8 @@ function getTile(source, tile, done) {
 
   tileRequest(url, function(err, res, body) {
     if (err) return done(err);
+    else if (res.statusCode === 200) return done(null, parseVT(body, tile, source));
+    else if (res.statusCode === 404) return done(null, parseVT(null, tile, source));
```
Comment (mourner) [inline src/remote.js]: I think we should handle this the same as in MBTiles, not doing any parsing and passing `undefined` as the reduce value which makes the main thread not fire a `reduce` event. https://github.com/mapbox/tile-reduce/blob/nextgen/src/mbtiles.js#L33

**Comment by mourner** [comment]
To be able to release 3.0 confidently before the end of the week, ideally we would settle for a much simpler solution, and then look into complex refactoring in subsequent releases.

**Comment by morganherlocker** [comment]
I refactored this to a different, simpler approach https://github.com/mapbox/tile-reduce/commit/7e8acf38a0dcea663ce42a4523770f27a2d4d2a0

**Pull request closed without merging**